### PR TITLE
Updating versions

### PR DESCRIPTION
**Pull request by juliovalcarcel**
Updating Spring Cloud versions as well as MongoDB patch version. 

Bumping to 1.0.0.BUILD-SNAPSHOT

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -24,7 +24,7 @@ env:
     - MONGODB=3.2.20
     - MONGODB=3.4.16
     - MONGODB=3.6.6
-    - MONGODB=4.0.0
+    - MONGODB=4.0.12
```
Comment (venilnoronha) [inline .travis.yml]: Unfortunately, due to the [license changes](https://www.mongodb.com/community/licensing) in MongoDB, I don't think we can do this at the moment. So I suggest that we revert this.

File: README.md [hunk 2]
```diff
@@ -12,7 +12,7 @@ Configure pom.xml, like this:
     <dependency>
         <groupId>org.springframework.cloud</groupId>
         <artifactId>spring-cloud-config-server-mongodb</artifactId>
-        <version>0.0.3.BUILD-SNAPSHOT</version>
+        <version>1.0.0.BUILD-SNAPSHOT</version>
```
Comment (venilnoronha) [inline README.md]: I think we're not ready for a 1.0.0 yet. Let's have this as 0.0.4.BUILD-SNAPSHOT.

File: pom.xml [hunk 3]
```diff
@@ -5,7 +5,7 @@
 
 	<groupId>org.springframework.cloud</groupId>
 	<artifactId>spring-cloud-config-server-mongodb</artifactId>
-	<version>0.0.3.BUILD-SNAPSHOT</version>
+	<version>1.0.0.BUILD-SNAPSHOT</version>
```
Comment (venilnoronha) [inline pom.xml]: Ditto.

**Review by venilnoronha — COMMENTED** [review]
Thanks for updating these libraries!

File: .travis.yml (hunk 1 above)
Comment (juliovalcarcel) [reply .travis.yml:27]: @venilnoronha with some time past now and also seeing that spring-data-mongo uses 4.2 in their [Dockerfile](https://github.com/spring-projects/spring-data-mongodb/blob/master/ci/openjdk11-mongodb-4.2/Dockerfile) is this something you would reconsider integrating if I updated this pull request with the latest mongo 4.0 and 4.2 versions? 

Additionally I can update the other spring cloud versions to latest were applicable and address the 0.0.4 issue

**Pull request closed without merging**